sktime/experiments/analysis.py

```diff
@@ -0,0 +1,374 @@
+
+import pandas as pd
+import numpy as np
+import itertools
+
+from scipy import stats
+from scipy.stats import ttest_ind
+from scipy.stats import ranksums
+
+from sklearn.metrics import accuracy_score, mean_squared_error
+import scikit_posthocs as sp
+
+from sktime.experiments.losses import Losses
+
+import matplotlib.pyplot as plt
+
+class AnalyseResults(object):
+    """
+    Analyze results of machine learning experiments.
+
+    Parameters
+    ----------
+    result: sktime result object
+        class for storing the results
+    """
+
+    def __init__(self, 
+                 results):
+
+        self._results_list = results.load()
+    
+    def prediction_errors(self, metric):
+        """
+        Calculates the average prediction error per estimator as well as the prediction error achieved by each estimator on individual datasets.
+
+        Parameters
+        -----------
+        metric: `sktime.analyse_results.scores`
+            Error function 
+        Returns
+        --------
+        pickle of pandas DataFrame
+            ``estimator_avg_error`` represents the average error and standard deviation achieved by each estimator. ``estimator_avg_error_per_dataset`` represents the average error and standard deviation achieved by each estimator on each dataset.
+        """
+        #load all predictions
+        losses = Losses(metric)
+        for res in self._results_list:
+
+            y_pred = res.y_pred
+            y_pred = list(map(float, y_pred))
+            y_true = res.y_true
+            y_true = list(map(float, y_true))
+
+            losses.evaluate(predictions=y_pred, 
+                            true_labels=y_true,
+                            dataset_name=res.dataset_name,
+                            strategy_name=res.strategy_name)
+        return losses.get_losses()
+
+
+    def average_and_std_error(self, scores_dict):
+        """
+        Calculates simple average and standard error.
+
+        Args:
+            scores_dict(dictionary): Dictionary with estimators (keys) and corresponding prediction accuracies on different datasets.
+        
+        Returns:
+            pandas DataFrame
+        """
+        result = {}
+        for k in scores_dict.keys():
+            average = np.average(scores_dict[k])
+            n = len(scores_dict[k])
+            std_error = np.std(scores_dict[k])/np.sqrt(n)
+            result[k]=[average,std_error]
+        
+        res_df = pd.DataFrame.from_dict(result, orient='index')
+        res_df.columns=['avg_score','std_error']
+        res_df = res_df.sort_values(['avg_score','std_error'], ascending=[1,1])
+
+        return res_df.round(3)
```

Comment:
I wouldn't round results internally, and leave it to the user to decide whether or not to round the returned output
```suggestion
        return res_df
```
Same applies to other methods

Reply:
ok, will push shortly, removing the rounding.